### PR TITLE
Property Set Apigee Sample

### PR DESCRIPTION
**Pull request by JoelGauci**
New Apigee Accelerator sample: Property Set
-  Basic implementation of Property Sets in Apigee
- JavaScript and Assign Message policies used to demonstrate Property Set usage
- Flow variables related to Property Set

### CODE REVIEW

**Comment by ssvaidyanathan** [comment]
@JoelGauci - can you please resolve the README.md conflict?

File: property-set/README.md [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+
+## (QuickStart) Setup using CloudShell
+
+Use the following GCP CloudShell tutorial, and follow the instructions.
+
+[![Open in Cloud Shell](https://gstatic.com/cloudssh/images/open-btn.png)](https://ssh.cloud.google.com/cloudshell/open?cloudshell_git_repo=https://github.com/GoogleCloudPlatform/apigee-samples&cloudshell_git_branch=main&cloudshell_workspace=.&cloudshell_tutorial=property-set/docs/cloudshell-tutorial.md)
+
+## Setup instructions
+
+1. Clone the `apigee-samples` repo, and switch the `property-set` directory
```
Comment (ssvaidyanathan) [inline property-set/README.md:101]: and switch "to" the `property-set` directory

File: property-set/apiproxy/proxies/default.xml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+-->
+<ProxyEndpoint name="default">
+  <Description/>
+  <FaultRules/>
+  <PreFlow name="PreFlow">
+    <Request>
+      <Step>
```
Comment (ssvaidyanathan) [inline property-set/apiproxy/proxies/default.xml:19]: can we move these policies to the <Response> flow as these are policies that are setting the response object?

File: property-set/apiproxy/proxies/default.xml [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+  </PreFlow>
+  <PostFlow name="PostFlow">
+    <Request/>
+    <Response/>
+  </PostFlow>
+  <Flows/>
+  <HTTPProxyConnection>
+    <BasePath>/v1/samples/property-set</BasePath>
+    <Properties/>
+    <VirtualHost>default</VirtualHost>
```
Comment (ssvaidyanathan) [inline property-set/apiproxy/proxies/default.xml:36]: VirutalHost not needed

File: property-set/docs/cloudshell-tutorial.md [hunk 4]
```diff
@@ -0,0 +1,127 @@
+# Property Set
+
+---
+This sample shows how to easily get data from a Property Set using an AssignMessage policy.
```
Comment (ssvaidyanathan) [inline property-set/docs/cloudshell-tutorial.md:4]: The same also gets data using JS policy. 
So lets copy the same description from the samples README here as well

File: property-set/docs/cloudshell-tutorial.md [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+
+```sh
+curl -v https://$APIGEE_HOST/v1/samples/property-set
+```
+
+Use the debug tool inside Apigee to see how we access values in a property set, using the following syntax:
+
+```propertyset.<property_set_name>.<property_name>```
+
+### AssignMessage policy to access a Property Set
```
Comment (ssvaidyanathan) [inline property-set/docs/cloudshell-tutorial.md:60]: I see a mismatch in the contents between the sample README and this tutorial. In this md file, we have this section thats missing in the sample README.

File: property-set/docs/cloudshell-tutorial.md [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+...and the JSON payload (response) that is created:
+
+```json
+{
+  "foo": "bar",
+  "message": "This is a basic message."
+}
+```
+
+Note: private information should be handled using encrypted
```
Comment (ssvaidyanathan) [inline property-set/docs/cloudshell-tutorial.md:91]: Lets add this to the main sample README as well which is missing

File: property-set/README.md (hunk 1 above)
Comment (JoelGauci) [reply property-set/README.md:101]: OK

File: property-set/apiproxy/proxies/default.xml (hunk 2 above)
Comment (JoelGauci) [reply property-set/apiproxy/proxies/default.xml:19]: Done

File: property-set/apiproxy/proxies/default.xml (hunk 3 above)
Comment (JoelGauci) [reply property-set/apiproxy/proxies/default.xml:36]: yes of course! - removed

File: property-set/docs/cloudshell-tutorial.md (hunk 4 above)
Comment (JoelGauci) [reply property-set/docs/cloudshell-tutorial.md:4]: Done

File: property-set/docs/cloudshell-tutorial.md (hunk 6 above)
Comment (JoelGauci) [reply property-set/docs/cloudshell-tutorial.md:91]: Done

File: property-set/README.md [hunk 7]
```diff
@@ -4,7 +4,7 @@ This sample shows how to easily define a Property Set and how to
 access data from it.
 This sample leverages the following policy:
 
-* [Assign Message](https://cloud.google.com/apigee/docs/api-platform/reference/policies/assign-message-policy?hl=en) policy to set the JSON response with values retrieved from a [property set](https://cloud.google.com/apigee/docs/api-platform/cache/property-sets)
+* [Assign Message](https://cloud.google.com/apigee/docs/api-platform/reference/policies/assign-message-policy?hl=en) policy to set the JSON response with values retrieved from a[property set](https://cloud.google.com/apigee/docs/api-platform/cache/property-sets)
```
Comment (ssvaidyanathan) [inline property-set/README.md:7]: space missing between "a" and "[property set]"

Comment (JoelGauci) [reply property-set/README.md]: Done

**Pull request merged**